waterloo-rocketry/minerva-rewrite
# Implement `/meeting-reminder` command

## Description

This PR implements the `/meeting-reminder` command, formerly known as `/meeting reminder`. This allows users to trigger a reminder for the soonest upcoming event in the channel that it was executed in. In addition, executing `/meeting-reminder ping` not only triggers a reminder, but also pings `@channel`.



This PR closes #22.

## Developer Testing

Testing done:
- Updated unit tests for `generateEventReminderChannelText` to handle the manually triggered cases
- Tested functionality on the dev slack, including its handling of error cases when trying to run the command in channels where minerva does not exist.

## Reviewer Testing



Steps to view/test 
1. Run the `/meeting-reminder` command in a channel that does not have any upcoming events in the next 24 hours and ensure that you get a message stating such
2. Run the `/meeting-reminder` command in a channel that has an upcoming event and ensure that a reminder message is sent
3. Run the `/meeting-reminder ping` command in a channel that has an upcoming event and ensure that a reminder message is sent and it contains and it pings `@channel`


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/minerva-rewrite/67)
